src/main/java/com/CezaryZal/manager/db/service/UserServiceImp.java

```diff
@@ -1,30 +1,35 @@
 package com.CezaryZal.manager.db.service;
 
 import com.CezaryZal.entity.User;
+import com.CezaryZal.exceptions.IncorrectLoginOrPasswordSecurityException;
 import com.CezaryZal.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.Optional;
+import java.util.List;
 
 @Service
 public class UserServiceImp implements UserService {
 
     private UserRepository userRepository;
 
+    @Autowired
     public UserServiceImp(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
 
-    public Optional<User> findById(Long index) {
-        return userRepository.findById(index);
+    public User findById(Long index) {
+        return userRepository.findById(index)
+                .orElseThrow(() -> new IncorrectLoginOrPasswordSecurityException("Login or password incorrectly entered"));
```

Comment:
bardziej rzuciłbym tutaj włąsnie jakiś wspomniany wyżej usernotfoundexception,
Sięgasz do repo po ID usera i jak go nie znajdziesz to rzucasz IncorrectLoginOrPasswordSecurityException. Wydaje mi się że to nie jest warstwa na ten exception.

Reply:
W przypadku dostępu tylko administratrora można szczegółowiej okreslić dlaczego nie mamy dostępu do zasobów ale przy warstwie użytkownika wspominałeś że powinna być jeden ogólny wyjątek, kwestie bezpieczeństwa.

Reply:
Ale czy wyjątek który rzucasz jest prawdziwy?
Szukasz użytkownika po ID i go nie znajdujesz.
Co do ma  tego ma login użytkownika?
Co do ma tego hasło użytkownika?